### DISCUSSION TITLE
Proper way to backup Dolby Vision from Blu-Ray and re-insert it into a re-encode of the blu-ray?

### DISCUSSION THREAD

**Discussion by advarksauce** [quietvoid/dovi_tool#215] [Q&A] [answered]
I rip my blu-rays with MakeMKV to preserve the Dolby Vision. And I want to re-encode the mkv with handbrake to reduce file size. But I also want to preserve all the Dolby Vision information from the makemkv file into the new encoded file.

I think I read that no hardware streaming players like the nvidia shield are able to process the EL.hevc properly and will only use the RPU data from it. But I figure maybe in the future this will be worked out with a new model hopefully, so I thought I would just back up the EL.hevc now with the hope that it could be fully utilized in future hardware.

Questions:
1. When I use _"dovi_tool demux makeMKVVideo.hevc"_ and it creates EL.hevc and BL.hevc, is the RPU information also included in the EL.hevc? Or do I need to separately use the _extract_ command to extract the RPU from the hevc stream so that I may _inject_ that into my re-encode? Or does muxing the EL.hevc with the encode.hevc include the RPU in there automatically?

2. Am I able to convert blu-ray rips to Profile 8.4? My understanding is that Blu-Rays are encoded in Profile 7 dolby vision, and that means the base layer is HDR (BT.2020). Does the _"dovi_tool -m 4 convert"_ command convert the base layer into HLG format so that it can play on both SDR and HDR monitors correctly?

Thank you for your time and appreciate any help and guidance you may offer.

**Comment by quietvoid** [comment, marked as answer]
1. Yes the RPU is in the EL file.
2. `dovi_tool` doesn't modify the image, so don't do that.

To be honest, as you use HandBrake you should look into getting a nightly build.
HandBrake supports Dolby Vision encoding from MakeMKV rips now.
But only in the yet to be released 1.7 version.

It'll be much simpler.

I don't think a non-disc player is ever going to properly support the EL.
And even then, the original EL would not be compatible with the reencoded video.

**Reply by advarksauce** [reply]
Thanks for the quick answer I'll be sure to check out the handbrake nightly! So in my case I only need to use the extract and inject rpu commands because there is no need for me to demux/mux the enhancement layer.

Just curious why wouldn't the original EL be compatible with an encode I thought it was just a video layer of additional colors that dolbyvision subtracts/adds to the original base layer?

**Reply by quietvoid** [reply]
> So in my case I only need to use the extract and inject rpu commands because there is no need for me to demux/mux the enhancement layer.

You don't need to do anything, HandBrake handles the RPU conversions and muxing.

> Just curious why wouldn't the original EL be compatible with an encode I thought it was just a video layer of additional colors that dolbyvision subtracts/adds to the original base layer? 

It's meant to be applied to the encoded base layer. The merging into any other video will likely cause issues.

**Reply by advarksauce** [reply]
Got it, thank you again for answering my questions!
